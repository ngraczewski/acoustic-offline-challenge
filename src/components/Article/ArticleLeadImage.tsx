import React from 'react';

export type Props = {
  imageSrc: string;
  imageAltText: string;
};

export const ArticleLeadImage = ({
  imageAltText,
  imageSrc,
}: Props): JSX.Element => {
  return (
    <img
      className="article__lead-image"
      alt={imageAltText}
      src={imageSrc}
    ></img>
  );
};
