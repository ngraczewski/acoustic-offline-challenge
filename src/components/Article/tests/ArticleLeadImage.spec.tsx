import React from 'react';
import faker from 'faker';
import { shallow } from 'enzyme';
import { ArticleLeadImage, Props } from '../ArticleLeadImage';

describe('ArticleLeadImage', () => {
  beforeEach(() => {
    faker.seed(1);
  });

  it('should render correctly', () => {
    const props: Props = {
      imageAltText: faker.lorem.sentence(),
      imageSrc: faker.internet.url(),
    };
    const wrapper = shallow(<ArticleLeadImage {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
