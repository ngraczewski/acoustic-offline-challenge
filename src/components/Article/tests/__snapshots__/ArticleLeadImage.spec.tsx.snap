// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleLeadImage should render correctly 1`] = `
<img
  alt="Repellat quisquam recusandae alias consequuntur corporis."
  className="article__lead-image"
  src="https://caesar.biz"
/>
`;
